redmine_import_time_entry/unit/backend_adapter.py

```diff
@@ -0,0 +1,103 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2015 - Present Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.tools.translate import _
+from openerp.addons.connector.exception import (
+    NothingToDoJob, InvalidDataError)
+from redmine import exceptions
+from openerp.addons.connector_redmine.unit.backend_adapter import \
+    RedmineAdapter
+from openerp.addons.connector_redmine.backend import redmine13
+
+
+@redmine13
+class TimeEntryAdapter(RedmineAdapter):
+    """
+    Time Entry Backend Adapter for Redmine
+
+    Time entries in Redmine can not be querried by the date of creation
+    or update. The alternative is to search for all entries for a
+    period of time and then filtering these by the field updated_on.
+    """
+
+    _model_name = 'redmine.hr.analytic.timesheet'
+
+    def search(self, updated_from, filters):
+        """
+        Get all time entry that were updated between the interval of time
+        """
+        self._auth()
+
+        if updated_from:
+            return [
+                entry.id for entry in
+                self.redmine_api.time_entry.filter(**filters)
+                if updated_from < entry.updated_on
+            ]
+        else:
+            # If the time entries are imported for the first time
+            # no need to filter by the date of update
+            return [
+                entry.id for entry in
+                self.redmine_api.time_entry.filter(**filters)
+            ]
+
+    def read(self, redmine_id):
+        self._auth()
+
+        try:
+            entry = self.redmine_api.time_entry.get(redmine_id)
+        except exceptions.ResourceNotFoundError:
+            raise NothingToDoJob(
```

Comment:
I found that `NothingToDoJob` was not a very good idea and it should hardly be used.
The issue is that it will cancel an entire job, so imagine a case where you import `A` which imports its dependencies `B` and `C`. `C` raises `NothingToDoJob` which cancels the whole job thus cancels the import of `A` as well.
I think it should just return `None` which has to be handled upstream.